.github/CONTRIBUTOR_GUIDE/CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+
+### Why Coding Standards Matter
+
+Coding standards are crucial for several reasons:
+
+1. **Consistency**: Consistent code is easier to read, understand, and maintain. It helps new team members quickly get up to speed and reduces the learning curve.
+2. **Readability**: Clear and well-structured code is easier to debug and extend. It allows developers to quickly identify and fix issues.
+3. **Maintainability**: Code that follows a standard structure is easier to refactor and update. It ensures that changes can be made with minimal risk of introducing new bugs.
+4. **Collaboration**: When everyone follows the same standards, it becomes easier to collaborate on code. It reduces friction and misunderstandings during code reviews and merges.
+
+### Scope of These Documents
+
+These documents cover the coding standards for the following types of files in our project:
+
+- **`install/$AppName-install.sh` Scripts**: These scripts are responsible for the installation of applications.
+- **`ct/$AppName.sh` Scripts**: These scripts handle the creation and updating of containers.
+- **`json/$AppName.json`**: These files store structured data and are used for the website.
+
+Each section provides detailed guidelines on various aspects of coding, including shebang usage, comments, variable naming, function naming, indentation, error handling, command substitution, quoting, script structure, and logging. Additionally, examples are provided to illustrate the application of these standards.
+
+By following the coding standards outlined in this document, we ensure that our scripts and JSON files are of high quality, making our project more robust and easier to manage. Please refer to this guide whenever you create or update scripts and JSON files to maintain a high standard of code quality across the project. 📚🔍
+
+Let's work together to keep our codebase clean, efficient, and maintainable! 💪🚀
+
+
+## Getting Started
+
+Before contributing, please ensure that you have the following setup:
+
+1. **Visual Studio Code** (recommended for script development)
+2. **Recommended VS Code Extensions:**
+   - [Shell Syntax](https://marketplace.visualstudio.com/items?itemName=bmalehorn.shell-syntax)
+   - [ShellCheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck)
+   - [Shell Format](https://marketplace.visualstudio.com/items?itemName=foxundermoon.shell-format)
+
+### Important Notes
+- Use [AppName.sh](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/ct/AppName.sh) and [AppName-install.sh](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/install/AppName-install.sh) as templates when creating new scripts.
+- The call to `community-scripts/ProxmoxVE` should be adjusted to reflect the correct fork URL.
```

Comment:
I like the idea to start a new script from a template.
Why not going all in and providing the CLI commands for this?

```
If you plan to implement a new application script, use our templates to get a kickstart:

$ NEW_APP_NAME="my-script"
$ cp ct/AppName.sh ct/${NEW_APP_NAME}.sh
$ cp ...
$ sed ... (some git remote -v magic to get the fork path)
```

This way, folks can just copy / paste the commands and get going.

Additionally: Considering the CONTIRBUTING.md is the entry doc, this is the first time it is mentioned that we operate on a fork.
Maybe it is better to move this chapter to a 

> Guide: Create your own App Script

which starts with

1. Fork this Repository
2. Kickstart your app with your template
3. ...

Reply:
There has been a proposal by @quantumryuu  for this already, have a look here:  https://github.com/community-scripts/ProxmoxVE/tree/new_script_testing/.github/CONTRIBUTOR_GUIDE/dev-scripts
I will start testing, experementing with this when the hollidays are over. 

I also would love to have something simillar to [this](https://community-scripts.github.io/ProxmoxVE/json-editor) for Appname.sh and AppName-install.sh

Reply:
About the AppName.sh/AppName-install.sh we could modify the scripts I proposed into asking things like "Author", "Source", "AppName" I guess.